components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/utils/grpc/GrpcGen/BalFileGenerationUtils.java

```diff
@@ -0,0 +1,230 @@
+     *
+     * @param exePath        protoc executor path
+     * @param protoPath      .proto file path
+     * @param descriptorPath file descriptor path.
+     * @return byte array of generated proto file.
+     */
+    public static byte[] getProtoByteArray(String exePath, String protoPath, String descriptorPath) {
+
+        String command = new ProtocCommandBuilder
+                (exePath, protoPath, resolveProtoFloderPath(protoPath), descriptorPath).build();
+        generateDescriptor(command);
+        File initialFile = new File(descriptorPath);
+        try (InputStream targetStream = new FileInputStream(initialFile)) {
+            DescriptorProtos.FileDescriptorSet set = DescriptorProtos.FileDescriptorSet.parseFrom(targetStream);
+            if (set.getFileList().size() > 0) {
+                return set.getFile(0).toByteArray();
+            }
+        } catch (IOException e) {
+            throw new BalGenToolException("Error reading generated descriptor file '" + descriptorPath + "'.", e);
+        }
+        return new byte[0];
+    }
+
+    /**
+     * Used to clear the temporary files.
+     *
+     * @param file file to be deleted
+     */
+    public static void delete(File file) {
+        if ((file != null) && file.exists() && file.isDirectory()) {
+            String files[] = file.list();
+            if (files != null) {
+                if (files.length != 0) {
+                    for (String temp : files) {
+                        File fileDelete = new File(file, temp);
+                        if (fileDelete.isDirectory()) {
+                            delete(fileDelete);
+                        }
+                        if (fileDelete.delete()) {
+                            LOG.debug("Successfully deleted file " + file.toString());
+                        }
+                    }
+                }
+            }
+            if (file.delete()) {
+                LOG.debug("Successfully deleted file " + file.toString());
+            }
+            if ((file.getParentFile() != null) && (file.getParentFile().delete())) {
+                LOG.debug("Successfully deleted parent file " + file.toString());
+            }
+        } else if (file != null) {
+            if (file.delete()) {
+                LOG.debug("Successfully deleted parent file " + file.toString());
+            }
+        }
+    }
+
+    /**
+     * Sae generated intermediate files.
+     *
+     * @param url  file URL
+     * @param file destination file location
+     */
+    public static void saveFile(URL url, String file) {
+        try (InputStream in = url.openStream(); FileOutputStream fos = new FileOutputStream(new File(file))) {
+            int length;
+            byte[] buffer = new byte[1024]; // buffer for portion of data from
+            while ((length = in.read(buffer)) > -1) {
+                fos.write(buffer, 0, length);
+            }
+            fos.close();
+            in.close();
```

Comment:
Shouldn't this,
1. fos.close();
2. in.close();
go in the "finally" block?

Reply:
this was a try-resource block doesn't need to close it get auto close.
we can remove below lines